src/main/kotlin/com/chattriggers/ctjs/minecraft/wrappers/Player.kt

```diff
@@ -222,26 +229,49 @@ object Player : CTWrapper<ClientPlayerEntity?> {
     @JvmStatic
     fun getActivePotionEffects(): List<PotionEffect> = toMC()?.activeStatusEffects?.values?.map(::PotionEffect).orEmpty()
 
+    // TODO(breaking): Return null instead of air BlockType for a miss
     /**
      * Gets the current object that the player is looking at,
-     * whether that be a block or an entity. Returns an air [BlockType] when not looking
+     * whether that be a block or an entity. Returns null when not looking
      * at anything.
      *
-     * @return the [Block], [Entity], [Sign], or [BlockType] being looked at
+     * @return the [Block] or [Entity] being looked at, or null if air
      */
     @JvmStatic
-    fun lookingAt(): Any {
-        TODO()
+    fun lookingAt(): Any? {
+        val target = Client.getMinecraft().crosshairTarget
+
+        return when (target?.type) {
+            HitResult.Type.MISS -> null
+            HitResult.Type.BLOCK -> {
+                val block = target as BlockHitResult
+                World.getBlockAt(BlockPos(block.blockPos)).withFace(BlockFace.fromMC(block.side))
+            }
+            HitResult.Type.ENTITY -> {
+                Entity.fromMC((target as EntityHitResult).entity)
+            }
+            null -> null
+        }
     }
 
     /**
-     * Gets the current item in the player's main hand.
+     * Gets the current item in the player's hand.
      *
+     * @param mainHand whether to get the item in the main hand or offhand
      * @return the current held [Item]
      */
+    @JvmOverloads
     @JvmStatic
-    fun getHeldItem(): Item? = toMC()?.inventory?.selectedSlot?.let {
-        getInventory()?.getStackInSlot(it)
+    fun getHeldItem(mainHand: Boolean = true): Item? {
```

Comment:
Might be nice to add an overload that accepts `Hand` instead, which would pair nicely with the recently changed you made to the interaction trigger

Reply:
Should I add Hand into providedlibs then?

Reply:
Yeah I guess that makes sense. I was hesitant about putting more stuff in the global scope, but it looks like Rhino will let you override global declarations with `var` so it's fine